tests/acceptance/features/bootstrap/NotificationContext.php

```diff
@@ -318,6 +318,15 @@ public function filterResponseByNotificationSubjectAndResource(string $subject,
 	 */
 	public function userShouldGetANotificationWithMessage(string $user, string $subject, TableNode $table):void {
 		$this->userListAllNotifications($user);
+		$this->featureContext->theHTTPStatusCodeShouldBe(200);
+		// sometimes the test might try to get notification before the notification is created by the server
+		// in order to prevent test from failing because of that list the notifications again
+		if (!isset($this->filterResponseAccordingToNotificationSubject($subject)->message)) {
+			\sleep(1);
+			$this->featureContext->setResponse(null);
+			$this->userListAllNotifications($user);
+			$this->featureContext->theHTTPStatusCodeShouldBe(200);
+		}
```

Comment:
can we use loop instead ? or 1 sec is pretty much enough to get the response. Also there is duplication of code.

Reply:
I think 1 sec is enough but I implemented the loop as you suggested.